swiftspec/core.py

```diff
@@ -142,18 +142,34 @@ def get_tokens_from_env(self):
         token = os.environ.get("OS_AUTH_TOKEN")
         url = os.environ.get("OS_STORAGE_URL")
         if token and url:
-            return [{"token": token, "url": url}]
+            return [{"X-Auth-Token": token, "url": url}]
+
+        sig = os.environ.get("TEMP_URL_SIG")
+        expire = os.environ.get("TEMP_URL_EXPIRES")
+        prf = os.environ.get("TEMP_URL_PREFIX")
+
+        if url and sig and expire:
+            return [{"url":url,"temp_url_sig": sig, "temp_url_expires": expire, "temp_url_prefix":prf}]
         else:
             return []
 
     def headers_for_url(self, url):
         headers = {}
         for auth in self.auth:
-            if url.startswith(auth["url"]):
-                headers["X-Auth-Token"] = auth["token"]
+            if url.startswith(auth["url"]) and "X-Auth-Token" in auth:
+                headers["X-Auth-Token"]=auth["X-Auth-Token"]
                 break
         return headers
 
+    def params_for_url(self, url):
+        params = {}
+        for auth in self.auth:
+            if url.startswith(auth["url"]) and "temp_url_sig" in auth:
```

Comment:
According to the [docs](https://docs.openstack.org/swift/latest/api/temporary_url_middleware.html), `temp_url_expires` is requred as well. Thus we might want to check for it at this place too.